enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -867,18 +864,27 @@ func (c *FirecrackerContainer) LoadSnapshot(ctx context.Context) error {
 		return err
 	}
 
+	ctx, cancel := c.monitorVMContext(ctx)
+	defer cancel()
+
 	err = (func() error {
 		_, span := tracing.StartSpan(ctx)
 		defer span.End()
 		span.SetName("StartMachine")
 		return machine.Start(vmCtx)
 	})()
 	if err != nil {
-		return status.InternalErrorf("Failed starting machine: %s", err)
+		if cause := context.Cause(ctx); cause != nil {
+			return cause
+		}
+		return status.InternalErrorf("failed to start machine: %s", err)
 	}
 	c.machine = machine
 
 	if err := c.machine.ResumeVM(ctx); err != nil {
+		if cause := context.Cause(ctx); cause != nil {
+			return cause
+		}
```

Comment:
These aren't strictly needed but I realized while testing that these can also fail due to UFFD failures, and without the `context.Cause` check, they return a generic "context cancelled" error. This makes them return the UFFD error which caused the VM to crash.

Having to check context.Cause everywhere is really ugly though so it'd be nice to find a design where we don't have to do this.